nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/reporting-tasks/edit-reporting-task/edit-reporting-task.component.ts

```diff
@@ -137,13 +144,16 @@ export class EditReportingTask {
         // build the form
         this.editReportingTaskForm = this.formBuilder.group({
             name: new FormControl(request.reportingTask.component.name, Validators.required),
-            state: new FormControl(request.reportingTask.component.state === 'STOPPED', Validators.required),
+            state: new FormControl(
+                { value: request.reportingTask.component.state !== 'DISABLED', disabled: this.readonly },
```

Comment:
- `readonly` is only supported by `input` and `textarea`. These are always set in the template.
- If the field we're attempting to disable is in the form directly (has the formControlName attribute) and you attempt to disable in the template, Angular will provide the following warning. This happens for components like a checkbox or any `ControlValueAccessor`. It is recommended that you enable and disable through the form API.

>   It looks like you're using the disabled attribute with a reactive form directive. If you set disabled to true
>   when you set up this control in your component class, the disabled attribute will actually be set in the DOM for
>   you. We recommend using this approach to avoid 'changed after checked' errors.
> 
>   Example:
>   // Specify the `disabled` property at control creation time:
>   form = new FormGroup({
>     first: new FormControl({value: 'Nancy', disabled: true}, Validators.required),
>     last: new FormControl('Drew', Validators.required)
>   });
> 
>   // Controls can also be enabled/disabled after creation:
>   form.get('first')?.enable();
>   form.get('last')?.disable();

- If the field your attempting to disable is not directly a form control (does not have the formControlName attribute - like an option in a mat-select) we disable through the template.

Reply:
Thanks for the clarification.